### DISCUSSION TITLE
Separating user-deployments

### DISCUSSION THREAD

**Discussion by saaramitay** [dagster-io/dagster#14382] [Q&A]
Hello, 

We are trying to separate the dagster-user-deployments from the centralised helm chart in order to achieve automatic deployment of pipelines with GitHub actions. 
So, in our ideal way each repo will have its own user-deployment chart and we only have to trigger the run to upgrade the deployments running in K8s on GCP, infrastructure of Dagster will still be deployed in the consolidated helm chart. 
For now we see that if we apply 2 different user-deployment charts in the same Dagster instance, it does not deploy 2 user-deployment pods but replaces the existing pod running (even tough it is different value.yaml files, different images and different names). 
Is there a way to separate the user-deployments into different helm charts in the way that i've mentioned?

**Comment by gibsondan** [comment]
Hi @saaramitay - we currently only support one user deployment chart for an open source dagster deployment. There's a feature request issue for moving to a more service discovery based model here that would make what you're describing possible I think: https://github.com/dagster-io/dagster/issues/6295

For now I think you will need a centralized CI/CD process unfortunately if you're using the Dagster helm chart.

**Reply by saaramitay** [reply]
Hi @gibsondan thanks for the answer. 
Do you have a guide or best-practice to implement dagster user-deployments with automated CI/CD? i've overviewed the documentation and i didn't see any related items

**Reply by mattfysh** [reply]
hey @gibsondan - is this still true? I noticed the `workspace.servers` value is an array suggesting perhaps more than one external workspace is supported?
